packages/engine/Source/Core/HeightmapTerrainData.js

```diff
@@ -94,16 +94,12 @@ import TerrainProvider from "./TerrainProvider.js";
  * @see GoogleEarthEnterpriseTerrainData
  */
 function HeightmapTerrainData(options) {
+  options = options ?? Frozen.EMPTY_OBJECT;
+
   //>>includeStart('debug', pragmas.debug);
-  if (!defined(options) || !defined(options.buffer)) {
-    throw new DeveloperError("options.buffer is required.");
-  }
-  if (!defined(options.width)) {
-    throw new DeveloperError("options.width is required.");
-  }
-  if (!defined(options.height)) {
-    throw new DeveloperError("options.height is required.");
-  }
+  Check.defined("options.buffer", options.buffer);
+  Check.defined("options.width", options.width);
+  Check.defined("options.height", options.height);
```

Comment:
Can we replace `Check.defined` with a more specific `typeOf` check where it makes sense while we're already updating these?